js/views/modals/purchase/Purchase.js

```diff
@@ -114,29 +130,46 @@ export default class extends BaseModal {
     launchSettingsModal({ initTab: 'Addresses' });
   }
 
+  clickApplyCoupon() {
+    this.coupons.addCode(this.$couponField.val());
+    this.$couponField.val('');
+  }
+
+  onKeyUpCouponCode(e) {
+    if (e.which === 13) {
+      this.coupons.addCode(this.$couponField.val());
+      this.$couponField.val('');
+    }
+  }
+
+  blurEmailAddress(e) {
+    this.order.set('alternateContactInfo', $(e.target).val());
+  }
+
+  blurMemo(e) {
+    this.order.set('memo', $(e.target).val());
+  }
+
+  changeCoupons() {
+    this.receipt.coupons = this.coupons.couponHashes;
```

Comment:
Since this is the only place you're using the couponHashes, a more robust way to get them here would be to pass them into the event rather than relying on a variable attached to the instance.

```
this.trigger('changeCoupons', this.couponHashes);

changeCoupons(couponHashes) { ... }
```